src/docfx/lib/markdown/validation/DocsValidationExtension.cs

```diff
@@ -231,5 +240,124 @@ static string GetContainerInlineContent(ContainerInline containerInline)
+                case ParagraphBlock:
+                    if (TryDetectTable((ParagraphBlock)node, markdownEngine))
+                    {
+                        tableNode = CreateValidationNode<TableNode>(isCanonicalVersion, node) with { IsSuccessParsed = false };
+                    }
+                    break;
+                default:
+                    break;
+            }
+            if (tableNode != null)
+            {
+                tableNodes.Add(tableNode);
+            }
+        }
+
+        private static TableNode TryParseTable(Table table)
+        {
+            if (table.Count == 0)
+            {
+                return new TableNode { IsSuccessParsed = false, };
+            }
+            var columnHeaders = new List<TableCellNode>();
+            var rowHeaders = new List<TableCellNode>();
+            var columnHeaderRow = (TableRow)table.First();
+            foreach (TableCell cell in columnHeaderRow)
+            {
+                columnHeaders.Add(ParseCell(cell));
+            }
+            foreach (TableRow row in table)
+            {
+                if (row.Count == 0)
+                {
+                    rowHeaders.Add(new TableCellNode { IsVisible = false });
+                }
+                else
+                {
+                    rowHeaders.Add(ParseCell((TableCell)row.First()));
+                }
+            }
+            return new TableNode
+            {
+                ColumnHeaders = columnHeaders.ToArray(),
+                RowHeaders = rowHeaders.ToArray(),
+                IsSuccessParsed = true,
+            };
+        }
+
+        private static TableCellNode ParseCell(TableCell cell)
+        {
+            if (cell == null || cell.Count == 0)
+            {
+                return new TableCellNode
+                {
+                    IsVisible = false,
+                    IsEmphasis = false,
+                };
+            }
+            var block = cell.LastChild;
+            var isVisible = MarkdigUtility.IsVisible(block);
+            var isEmphasis = false;
+            MarkdigUtility.Visit(block, node =>
+            {
+                var innerEmphasis = node switch
+                {
+                    ParagraphBlock paragraph when paragraph.Inline.FindDescendants<EmphasisInline>().Any() => true,
+                    _ => false,
+                };
+                return isEmphasis = isEmphasis || innerEmphasis;
+            });
+            var cellNode = new TableCellNode
+            {
+                IsVisible = isVisible,
+                IsEmphasis = isEmphasis,
+            };
+            return cellNode;
+        }
+
+        private static bool TryDetectTable(ParagraphBlock paragraph, MarkdownEngine markdownEngine)
+        {
+            if (!paragraph.Inline.FindDescendants<LineBreakInline>().Any())
+            {
+                return false;
+            }
+
+            var plainText = markdownEngine.ToPlainText(paragraph);
```

Comment:
`ToPlainText` for every paragraph with a `LineBreakInline` is a pretty expensive operation. Consider traverse the `ParagraphBlock` DOM to determine if this is likely a misconfigured table.

Performance Tips: minimize allocation for read operations.